app/components/debug-grids.js

```diff
@@ -44,6 +46,25 @@ export default class DebugGridsComponent extends Component {
 
   @tracked debugFeaturesData = {
     featureFlags: ['dev'],
-    isSuperUser: this.login.userData.roles?.super_user ?? false,
+    isSuperUser:
+      this.login.userData?.roles?.super_user ||
+      this.login.userData?.disabled_roles?.includes('super_user'),
   };
+
+  @action async toggleSuperUser({ target }) {
+    const response = await fetch(`${APPS.API_BACKEND}/users/self?dev=true`, {
+      method: 'PATCH',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ disabledRoles: ['super_user'] }),
+    });
+    if (response.ok) {
+      this.debugUserRolesData = {
+        ...this.debugUserRolesData,
+        super_user: target.checked,
+      };
+    }
+  }
```

Comment:
Currently, there's no fallback if the PATCH request fails. It would be a good idea to handle errors gracefully. You can add something like giving users proper feedback, either through a toast message or by toggling the checkbox back if the action fails.

Reply:
@Achintya-Chatterjee Addressed all the comments can you please re review.